### PR TITLE
Check primary duplicate by group

### PR DESCRIPTION
**Pull request by w-zr**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #8427

## What this PR does / why we need it:

Check primary duplicate by group on insertion.

### CODE REVIEW

File: pkg/sql/compile/compile.go [hunk 1]
```diff
@@ -357,32 +354,54 @@ func (c *Compile) compileApQuery(qry *plan.Query, ss []*Scope) (*Scope, error) {
...
 			arg.IsRemote = true
-			for _, scope := range ss {
-				scope.Instructions = append(scope.Instructions, vm.Instruction{
-					Op:  vm.PreInsert,
-					Arg: preArg,
-				})
-			}
-
-			rs = c.newInsertMergeScope(arg, preArg, ss)
+			rs = c.newInsertMergeScope(arg, ss)
```
Comment (JackTan25) [inline pkg/sql/compile/compile.go:404]: same with last comment. I think Insert should just appear after the group operator. And you also need a mergegroup operator to do duplicate-check

**Comment by JackTan25** [comment]
by the way, why not use agg.eval to achieve this, but write another func interanl_check_diplicate_one

**Comment by ouyuanning** [comment]
we have lots of case to insert batch. not only in Insert Operator.
this code only can check for normal insert clause.   i don't known if this is what you want.

**Comment by w-zr** [comment]
> we have lots of case to insert batch. not only in Insert Operator. this code only can check for normal insert clause. i don't known if this is what you want.

This is not completed. This pr only supports duplicate check in new inserted data.

**Pull request closed without merging**